diesel_cli/src/pretty_printing.rs

```diff
@@ -0,0 +1,194 @@
+
+
+/// Simple pretty printer hand tailored for the output generated
+/// by the `quote` - crate for schema inference
+///
+/// Rules:
+///   1. Seeing { increases indentation level
+///   2. Seeing } decreases indentation level
+///   3. Insert newline after {, }, ,, and ;
+///   4. Don't put spaces:
+///      between ident and !,
+///      between path segments and ::
+///      after ( and before )
+///      before ,
+pub fn format_schema<W>(schema: &str, mut output: W) where W: ::std::io::Write {
+    let mut out = String::new();
+    let mut indent = String::new();
+    let mut skip_space = false;
+    let mut last_char = ' ';
+    
+    for c in schema.chars() {
+        // quote inserts whitespaces at some strange location,
+        // remove them:        
+        match c {
+            '!' | '(' | ';' | ',' | '<' | ')' | '>' if last_char.is_whitespace()
+                => { out.pop(); }
+            ':' if last_char.is_whitespace() => {
+                out.pop();
+                match out.pop() {
+                    Some(c @ '>') => {
+                        out.push(c);
+                        out += " ";
+                    }
+                    Some(c) => {
+                        out.push(c);
+                    }
+                    _ => {}
+                }
+            }
+            _=> {}
+        }
+        if skip_space && c.is_whitespace() && last_char != '>' {
+            skip_space = false;
+            continue;
+        }
+        last_char = c;
+        skip_space = false;
+        
+        // there is already an empty line before {
+        // we need to remove the already inserted indent, because the
+        // new indent is smaller than the old one
+        if c == '}' {
+            while let Some(c) = out.pop() {
+                if c == '\n' {
+                    break;
+                }
+            }
+            indent.pop();
+            out += "\n";
+            out += &indent;
+        }
+        // push the current token to the output string
+        out = format!("{}{}", out, c);
+        
+        // we need to insert newlines in some places and adjust the indent
+        // also we need to remember if we could skip the next whitespace
+        match c {
+            ';' | ',' | '}'  => {
+                skip_space = true;
+                out += "\n";
+                out += &indent;
+            }
+            ':' | '(' => skip_space = true,
+            '{' => {
+                skip_space = true;
+                out += "\n";
+                indent += "\t";
+                out += &indent;
+            }
+            _=>{}
+        }        
+    }
+    write!(&mut output, "{}", out).unwrap();
```

Comment:
Can you make this function return `std::io::Result<()>` and omit the unwrap?

And now for something really profound: Tabs vs. spaces. I see you used `\t` for indentation, probably because it's exactly one character. Idiomatic Rust code uses four spaces, though. Can you add a `.replace("\t", "    ")` here?